### ISSUE TITLE
Plugin crashing on specific user interaction

### ISSUE THREAD

**Issue by oSumAtrIX** [Garulf/FlowYouTube#1]
# System Information

Flow Launcher version: 1.9.2
OS Version: Microsoft Windows NT 10.0.19044.0
x64: True

# Cause

After right-clicking on a query result, <kbd>esc</kbd>aping back to query list causes the program to throw an exception with the stack trace below.

## Stacktrace

```
Python Path: 
Date: 03/01/2022 04:56:47
Exception:
System.IO.InvalidDataException: FlowYouTube|Exception ignored in: <function Launcher.__del__ at 0x000002392B7A9D80>
Traceback (most recent call last):
  File "C:\Users\oSumAtrIX\AppData\Roaming\FlowLauncher\Plugins\FlowYouTube-0.1.2\lib\flox\launcher.py", line 59, in __del__
    self.__call__()
  File "C:\Users\oSumAtrIX\AppData\Roaming\FlowLauncher\Plugins\FlowYouTube-0.1.2\lib\flox\__init__.py", line 96, in __call__
    return super().__call__()
  File "C:\Users\oSumAtrIX\AppData\Roaming\FlowLauncher\Plugins\FlowYouTube-0.1.2\lib\flox\launcher.py", line 41, in __call__
    results = request_method(*request_parameters)
  File "C:\Users\oSumAtrIX\AppData\Roaming\FlowLauncher\Plugins\FlowYouTube-0.1.2\lib\flox\__init__.py", line 127, in _query
    self.query(query)
  File "C:\Users\oSumAtrIX\AppData\Roaming\FlowLauncher\Plugins\FlowYouTube-0.1.2\plugin\main.py", line 27, in query
    self._results = utils.cache(query, max_age=MAX_CACHE_AGE, dir=path)(self.search)(query)
  File "C:\Users\oSumAtrIX\AppData\Roaming\FlowLauncher\Plugins\FlowYouTube-0.1.2\lib\flox\utils.py", line 32, in wrapper
    cache = json.load(f, encoding='utf-8')
  File "P:\Programme (64 Bit)\Python\3.10.0\lib\json\__init__.py", line 293, in load
    return loads(fp.read(),
  File "P:\Programme (64 Bit)\Python\3.10.0\lib\json\__init__.py", line 359, in loads
    return cls(**kw).decode(s)
TypeError: JSONDecoder.__init__() got an unexpected keyword argument 'encoding'

   at Flow.Launcher.Core.Plugin.JsonRPCPlugin.ExecuteAsync(ProcessStartInfo startInfo, CancellationToken token) in C:\projects\flow-launcher\Flow.Launcher.Core\Plugin\JsonRPCPlugin.cs:line 308
   at Flow.Launcher.Core.Plugin.JsonRPCPlugin.ExecuteAsync(ProcessStartInfo startInfo, CancellationToken token) in C:\projects\flow-launcher\Flow.Launcher.Core\Plugin\JsonRPCPlugin.cs:line 308
   at Flow.Launcher.Core.Plugin.JsonRPCPlugin.QueryAsync(Query query, CancellationToken token) in C:\projects\flow-launcher\Flow.Launcher.Core\Plugin\JsonRPCPlugin.cs:line 323
   at Flow.Launcher.Core.Plugin.PluginManager.<>c__DisplayClass22_0.<<QueryForPluginAsync>b__0>d.MoveNext() in C:\projects\flow-launcher\Flow.Launcher.Core\Plugin\PluginManager.cs:line 192
--- End of stack trace from previous location ---
   at Flow.Launcher.Infrastructure.Stopwatch.DebugAsync(String message, Func`1 action) in C:\projects\flow-launcher\Flow.Launcher.Infrastructure\Stopwatch.cs:line 40
   at Flow.Launcher.Core.Plugin.PluginManager.QueryForPluginAsync(PluginPair pair, Query query, CancellationToken token) in C:\projects\flow-launcher\Flow.Launcher.Core\Plugin\PluginManager.cs:line 211
   at Flow.Launcher.ViewModel.MainViewModel.<>c__DisplayClass137_0.<<QueryResults>g__QueryTask|2>d.MoveNext() in C:\projects\flow-launcher\Flow.Launcher\ViewModel\MainViewModel.cs:line 623
--- End of stack trace from previous location ---
   at Flow.Launcher.ViewModel.MainViewModel.QueryResults() in C:\projects\flow-launcher\Flow.Launcher\ViewModel\MainViewModel.cs:line 624
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__140_0(Object state)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
```

**Comment by Garulf** [comment]
Ah! It seems the `encoding` was removed from the json module in Python version 3.9 and above. 

I'll have a fix out shortly. In the mean time you can work around this error by pointing to python 3.8.

**Comment by Garulf** [comment]
Hi @oSumAtrIX just wanted you to know this *should* be fixed in the latest version (v0.2.0)!
